elastic/detection-rules
# [FR] Add enhancements to `release-fleet` workflow

## Related
* https://github.com/elastic/detection-rules/pull/2605

## Summary
This pull request adds the following to the previously fixed [release fleet ](https://github.com/elastic/detection-rules/actions/workflows/release-fleet.yml) worklfow.

➕ Add a step to checkout a specific commit hash for the detection rules repository. This is a value supplied in the workflow dispatch and is required. This step is required because we always want to freeze the state of the rules for a specific branch, based on the lock-versions PR to ensure changes across packages are all related to the same minor release.

➕ Add steps that create a release tag in Detection Rules if "ga" is supplied as the value to `package_maturity`. We should only create an integration tag if the beta has been tested and tests completed. Since we do beta, then ga this assumes that the production package being released was thoroughly tested and the state of the detection rules repository and it's associated changes are ga.

<details>
<summary> Screenshots </summary>
<img width="1681" alt="Screenshot 2023-03-01 at 4 39 37 PM" src="https://user-images.githubusercontent.com/99630311/222272046-67096422-b529-4c3a-9285-d738af22095b.png">

</details>

## Testing
A test workflow has already been done and can be referenced [here](https://github.com/elastic/detection-rules/actions/runs/4308183453). Note that it was manually stopped by myself to avoid PR creation to integrations and because the steps added were successfully executed and showed the expected results.

1. Navigate to Actions
2. Select `release-fleet` workflow
3. Select `run-workflow`
4. Change `Use workflow from: main` to `Use workflow from: enhancement-release-fleet-workflow-commit-option`

⚠️ Since the PR is to main, it will use the `packages.yml` contents where the package version will be `8.8-beta.1`. This will also be used to create the integrations branch locally and create the PR. Therefore, if multiple people are testing, there may be conflicting names.

To solve this:
1. Go to `https://github.com/elastic/integrations/pulls` and find the corresponding PR
2. Close with `This is testing only by TRaDE regarding pre-built rules package automation`
3. Go to `https://github.com/elastic/integrations/branches` and delete the corresponding branch

Also, some bug fixes that were fixed in https://github.com/elastic/detection-rules/pull/2605 will not be applicable to the workflow when testing because we are checking out a commit that is older. The only issue here is you will not see the correct package version (will be `8.7.2` instead of `8.8.1`) when versions are bumped.

### Expected Results
Please see the screenshots above.

## Updated 03-06-2023
@brokensound77 had originally requested in a peer review that we add a check to ensure the commit message checked out was indeed that of a lock versions PR, generated by the lock-versions workflow. In order to accomplish this another job `check-commit` had to be created which checks the commit hash's message for `Lock version` and if so, sets an output of `true` as an accessible variable in the job. This can then be checked with an `if` statement in the `fleet-pr` job where if `true` run the `fleet-pr` job and if not skip it.

Reference: https://github.com/elastic/detection-rules/actions/runs/4349113453/jobs/7598396688

For testing, please follow the steps listed above and try a commit hash from a locked versions commit hash (c07ced2ce414972508399895dfe877e2f3255b30) and one that is not. You should see that the `fleet-pr` job is skipped if it is not correct.
